packages/docusaurus-theme-common/src/utils/docsPreferredVersion/useDocsPreferredVersion.ts

```diff
@@ -15,7 +15,10 @@ import {DEFAULT_PLUGIN_ID} from '@docusaurus/constants';
 // Note, the preferredVersion attribute will always be null before mount
 export function useDocsPreferredVersion(
   pluginId: string | undefined = DEFAULT_PLUGIN_ID,
-) {
+): {
+  readonly preferredVersion: any;
+  readonly savePreferredVersionName: (versionName: string) => void;
+} {
```

Comment:
do we really want to have explicit returns everywhere?

I find this a bit annoying personally and like to leverage inference in some cases